File: clients/ios/TmuxRemote/Models/PairingInfo.swift
import Foundation

struct PairingInfo {
    var productId: String
    var deviceId: String
    var username: String
    var password: String
    var sct: String

    /// Parse a pairing string: p=<product>,d=<device>,u=<user>,pwd=<pass>,sct=<token>
    static func parse(_ string: String) -> PairingInfo? {
        let trimmed = string.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmed.isEmpty else { return nil }

        var fields: [String: String] = [:]
        for part in trimmed.split(separator: ",") {
            let kv = part.split(separator: "=", maxSplits: 1)
            guard kv.count == 2 else { continue }
            fields[String(kv[0])] = String(kv[1])
        }

        guard let productId = fields["p"],
              let deviceId = fields["d"],
              let password = fields["pwd"],
              let sct = fields["sct"] else {
            return nil
        }
        let username = fields["u"] ?? "owner"
        guard !username.isEmpty else { return nil }

        return PairingInfo(
            productId: productId,
            deviceId: deviceId,
            username: username,
            password: password,
            sct: sct
        )
    }
}

File: clients/android/app/src/main/java/com/nabto/tmuxremote/models/PairingInfo.kt
package com.nabto.tmuxremote.models

data class PairingInfo(
    val productId: String,
    val deviceId: String,
    val username: String,
    val password: String,
    val sct: String
) {
    companion object {
        /** Parse a pairing string: p=<product>,d=<device>,u=<user>,pwd=<pass>,sct=<token> */
        fun parse(string: String): PairingInfo? {
            val trimmed = string.trim()
            if (trimmed.isEmpty()) return null

            val fields = mutableMapOf<String, String>()
            for (part in trimmed.split(",")) {
                val kv = part.split("=", limit = 2)
                if (kv.size == 2) {
                    fields[kv[0]] = kv[1]
                }
            }

            val productId = fields["p"] ?: return null
            val deviceId = fields["d"] ?: return null
            val password = fields["pwd"] ?: return null
            val sct = fields["sct"] ?: return null
            val username = fields["u"] ?: "owner"
            if (username.isEmpty()) return null

            return PairingInfo(
                productId = productId,
                deviceId = deviceId,
                username = username,
                password = password,
                sct = sct
            )
        }
    }
}
